test/user.test.js

```diff
@@ -166,23 +336,84 @@ describe("Update forum user successfully with valid ID", function() {
             .expect(201)
             .end(function(err, res) {
                 if (err) done(err);
+                const id = res.body.userData._id;
                 request(app)
-                    .patch(`/api/v1/users/${res.body.user._id}`)
-                    .send(
-                        {
-                            username: "OldBob",
-                            displayName: "Bob",
-                            email: "joe@bobby.com",
-                            password: "JoeBob"
-                        })
-                    .expect(201)
+                    .post('/api/v1/users/login')
+                    .send({
+                        username: 'NewBob',
+                        email: 'bobby@joe.com',
+                        password: 'BobbyJoe'
+                    })
+                    .expect(200)
                     .end(function(err, res) {
                         if (err) done(err);
-                        assert.equal(res.body.updatedForumUser.username, "OldBob");
-                        assert.equal(res.body.updatedForumUser.displayName, "Bob");
-                        assert.equal(res.body.updatedForumUser.email, "joe@bobby.com");
-                        assert.equal(res.body.updatedForumUser.hashedPassword, "JoeBob");
-                        done();
+                        request(app)
+                            .patch(`/api/v1/users/${id}`)
+                            .set({ "X-Authorization": res.body.authToken })
+                            .send(
+                                {
+                                    username: "OldBob",
+                                    displayName: "Bob",
+                                    email: "joe@bobby.com",
+                                    password: "JoeBob"
+                                })
+                            .expect(201)
+                            .end(function(err, res) {
+                                if (err) done(err);
+                                assert.equal(res.body.updatedForumUser.username, "OldBob");
+                                assert.equal(res.body.updatedForumUser.displayName, "Bob");
+                                assert.equal(res.body.updatedForumUser.email, "joe@bobby.com");
+                                assert.equal(res.body.updatedForumUser.hashedPassword, "JoeBob");
```

Comment:
This will need call the hashPassword method, as the password and the hashedPasswords won't match.

Reply:
The feature isn't yet implemented and I would assume whoever is implementing the password hash feature would apply necessary changes here